identity/webapp/pages/registration.tsx

```diff
@@ -88,8 +111,16 @@ const RegistrationPage: NextPage<Props> = ({ user: auth0UserClaims }) => {
     contextUser ||
     auth0UserProfileToUserInfo(auth0UserClaims as Auth0UserProfile);
 
-  const updateUser = (/* formData: RegistrationInputs  */) => {
-    // Send form data back to action that brought us to the registration page
+  const updateUser = (formData: RegistrationInputs) => {
+    if (typeof decodedToken !== 'string') {
+      const newToken = generateNewToken(decodedToken, auth0State, formData);
+
+      const url = `${redirectUri}?state=${auth0State}&session_token=${newToken}`;
+
+      router.push(url);
+    } else {
+      // TODO: ?
+    }
   };
```

Comment:
I wonder if we can do this in a less JavaScripty way - using the router always seems like a hack. For example, could the form have an `action` pointing at an API endpoint (added by us in `/pages/api/`) which then created the token and did the redirect with a `301`?

Reply:
Yeah, definitely nicer. Will update.